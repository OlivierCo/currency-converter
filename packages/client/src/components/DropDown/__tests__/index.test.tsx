import React from 'react'
import { shallow, mount } from 'enzyme'
import { DropDown } from '../index'

describe('DropDown', () => {
  it('should be defined', () => {
    expect(DropDown).toBeDefined()
  })

  it('should render correctly', () => {
    const wrapper = shallow(
      <DropDown name="SuperName">
        <div className="foo">Bar 1</div>
        <div className="foo">Bar 2 </div>
        <div className="foo">Bar 3</div>
      </DropDown>,
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('should render DropDown with children', () => {
    const wrapper = mount(
      <DropDown name="SuperName">
        <div className="foo">Bar 1</div>
        <div className="foo">Bar 2 </div>
        <div className="foo">Bar 3</div>
      </DropDown>,
    )

    expect(wrapper.children().length).toEqual(1)
  })

  it('should get name', () => {
    const wrapper = mount(
      <DropDown name="SuperName">
        <div className="foo">Bar 1</div>
      </DropDown>,
    )

    expect(wrapper.prop('name')).toEqual('SuperName')
  })
})
