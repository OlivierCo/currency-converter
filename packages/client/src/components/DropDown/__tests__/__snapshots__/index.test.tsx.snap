// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown should render correctly 1`] = `
<EmotionCssPropInternal
  __EMOTION_TYPE_PLEASE_DO_NOT_USE__={[Function]}
  css={
    Object {
      "map": undefined,
      "name": "y2yp78",
      "next": undefined,
      "styles": "position:relative;text-align:center;width:100%;",
      "toString": [Function],
    }
  }
>
  <Button
    onClick={[Function]}
  >
    SuperName
  </Button>
</EmotionCssPropInternal>
`;
